taskvine/src/manager/vine_manager.c

```diff
@@ -898,10 +898,13 @@ static void cleanup_worker_files(struct vine_manager *q, struct vine_worker_info
 		// may correspond to a cache-update of a file that has not been declared
 		// yet.
 		if (!f || !delete_worker_file(q, w, f->cached_name, f->cache_level, VINE_CACHE_LEVEL_WORKFLOW)) {
-			replica = vine_file_replica_table_remove(q, w, cached_name);
+			if(cached_name){
```

Comment:
I think the issue is in line 889. `cached_names` is keeping pointers to the actual table keys, which may may free at any moment. A `cached_names[i] = xxstrdup(cached_name);` and then a free on those should fix it, but we should look for a cleaner way.